src/components/ReportActionItem/MoneyRequestAction.js

```diff
@@ -66,11 +79,39 @@ const defaultProps = {
     iouReport: {},
     reportActions: {},
     isHovered: false,
+    session: {
+        email: null,
+    },
 };
 
 const MoneyRequestAction = (props) => {
     const hasMultipleParticipants = props.chatReport.participants.length > 1;
     const onIOUPreviewPressed = () => {
+        // This would ideally be passed as a prop or hooked up via withOnyx so that we are not be triggering a potentially intensive
+        // search in an onPress handler, I think this could lead to performance issues but it probably ok for now.
+        const thread = ReportUtils.getThreadForReportActionID(props.action.reportActionID);
+        console.log({thread});
+        if (_.isEmpty(thread)) {
+            // Since a thread does not exist yet then we need to create it now. This is done by creating the report object
+            // and passing the parentReportActionID of the reportAction. OpenReport will then automatically create the thread for us.
+            const optimisticThreadReport = ReportUtils.buildOptimisticChatReport(
+                props.chatReport.participants,
+                props.translate('iou.threadReportName', {
+                    formattedAmount: CurrencyUtils.convertToDisplayString(props.iouReport.iouReportAmount, props.iouReport.currency),
+                    comment: props.action.originalMessage.comment,
+                }),
+                '',
+                props.chatReport.policyID,
+                props.chatReport.owner,
+                props.chatReport.isOwnPolicyExpenseReport,
+                props.chatReport.oldPolicyName,
+                undefined,
+                CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+                props.action.reportActionID,
+            );
+            console.log({optimisticThreadReport});
+            Report.openReport(optimisticThreadReport.reportID, optimisticThreadReport.participants, optimisticThreadReport);
+        }
         if (hasMultipleParticipants) {
             Navigation.navigate(ROUTES.getReportParticipantsRoute(props.chatReportID));
```

Comment:
Note that I think in order to get the newly created thread to show up we'll need to have something like GetChats triggered OR navigate to the report here in this section. 

@Julesssss and @luacmartins had mentioned that this might be in the works else where as well but I'm not sure of a GH or anything for it. 

If nothing changes here with these routes then the modal will continue to open instead of going to the thread.